scripts/dependency_graph.py

```
@@ -0,0 +1,109 @@
+        for filename in filenames:
+            if filename in ("config.json", "config.json.disabled"):
+                with (dirpath / filename).open("r") as file:
+                    raw = file.read()
+                    data = json.loads(raw)
+                    results[data["appName"]] = {
+                        "dependencies": set(data.get("dependencies", [])),
+                        "file": (dirpath / filename)
+                    }
+    return results
+
+def make_groups(dependencies: dict[str, DependencyFile], group_count: int = 1, matched_deps: set[str] = set(), results: dict[int, set[str]] | None = None):
```

Ruff (0.14.4)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)